### PR TITLE
Feat/notifications

### PR DESCRIPTION
**Pull request by Blaxwo**
## Describe your changes

- I created notification page.
- I created sockets for sending notification.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](#https://2024internship.atlassian.net/browse/SCRUM-91?atlOrigin=eyJpIjoiZDg1NGJkOTRhODE4NGFmNDgxNGE2NDYyNmVhNTcxYzAiLCJwIjoiaiJ9)

### **General**

- [ ] Assigned myself to the PR
- [ ] Assigned the appropriate labels to the PR
- [ ] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [ ] Performed a self-review of my code
- [ ] Types for input and output parameters
- [ ] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [ ] Don't have magic numbers
- [ ] Compare only with constants not with strings
- [ ] No ternary operator inside the ternary operator
- [ ] Don't have commented code
- [ ] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [ ] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [ ] No hardcoded values
- [ ] covered by tests
- [ ] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [ ] Components and business logic are separated
- [ ] Colors, Font Size, and Font Name is on the theme or in the constants
- [ ] No text in the components, use i18n approach
- [ ] No inline styles
- [ ] Imports are absolute
- [ ] Attach a screenshot if PR has visual changes.
зображення

### CODE REVIEW

File: src/pages/Notifications/index.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          (notification: INotification, index: number) => (
+            <NotificationMassage key={index} sx={{ padding: '12px 16px' }}>
+              <NotificationHeader>
+                <NotificationTitle variant="subtitle1" fontWeight="bold">
+                  {notification.type}
+                </NotificationTitle>
+                <Typography variant="caption">
+                  {new Date(notification.createdAt).toLocaleTimeString(
+                    'en-US',
+                    { hour: '2-digit', minute: '2-digit', hour12: true }
```
Comment (djanbolotov) [inline src/pages/Notifications/index.tsx]: Date and time formats are on the constants

File: src/common/hooks/useNotificationSocket.ts
Comment (djanbolotov) [inline src/common/hooks/useNotificationSocket.ts:1]: connect to the socket when user is logged in. else don't connect. you can use autoconnect: false and check if user has access_token then connect

**Review by djanbolotov — APPROVED** [review]

File: src/pages/Notifications/index.tsx (hunk 1 above)
Comment (Blaxwo) [reply src/pages/Notifications/index.tsx]: fixed

File: src/common/hooks/useNotificationSocket.ts
Comment (Blaxwo) [reply src/common/hooks/useNotificationSocket.ts:1]: fixed

**Pull request merged**